### PR TITLE
ARROW-9052: [CI][MinGW] Enable Gandiva

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

File: ci/scripts/cpp_test.sh [hunk 1]
```diff
@@ -32,13 +32,36 @@ export LD_LIBRARY_PATH=${ARROW_HOME}/${CMAKE_INSTALL_LIBDIR:-lib}:${LD_LIBRARY_P
...
 case "$(uname)" in
   Linux)
     n_jobs=$(nproc)
     ;;
   Darwin)
     n_jobs=$(sysctl -n hw.ncpu)
     ;;
+  MINGW*)
+    n_jobs=${NUMBER_OF_PROCESSORS:-1}
+    exclude_tests="gandiva-internals-test"
```
Comment (kszucs) [inline ci/scripts/cpp_test.sh:45]: I assume we'll need a follow-up jira to fix these tests.

**Comment by kszucs** [comment]
Restarted the RTools builds because the errors looked transient.

File: ci/scripts/cpp_test.sh (hunk 1 above)
Comment (kou) [reply ci/scripts/cpp_test.sh:47]: Done: https://issues.apache.org/jira/browse/ARROW-9072

**Review by kszucs — APPROVED** [review]

File: ci/scripts/cpp_test.sh (hunk 1 above)
Comment (kszucs) [reply ci/scripts/cpp_test.sh:47]: Thanks!

**Comment by kszucs** [comment]
Merging on green.

**Pull request closed without merging**